mbed-drivers/v1/I2C.hpp

```diff
@@ -0,0 +1,562 @@
+        return _next;
+    }
+
+    /**
+     * Accessor for the Transactions's issuer
+     * @return the I2C object that issued this transaction
+     */
+    I2C * get_issuer() {
+        return _issuer;
+    }
+
+    /**
+     * Accessor for the current segment pointer
+     * @return the current segment poitner
+     */
+    detail::I2CSegment * get_current() {
+        return _current;
+    }
+
+    /**
+     * Accessor for the irqsafe flag
+     * @retval true if the transaction was instantiated with irqsafe set and access to irq-safe allocators
+     * @retval false if the transaction was not instantiated with irqsafe set and access to irq-safe allocators
+     */
+    bool is_irqsafe() const {
+        return _irqsafe;
+    }
+
+    /**
+     * Accessor for the transaction frequency
+     * @return the frequency of the transaction in Hz
+     */
+    uint32_t freq() const {
+        return _hz;
+    }
+
+    /**
+     * Accessor for the transaction frequency
+     * @param[in] hz the transaction frequency in Hz
+     */
+    void freq(uint32_t hz) {
+        _hz = hz;
+    }
+
+    /**
+     * Accessor for the transaction target address
+     * @return the transaction address
+     */
+    uint16_t address() const {
+        return _address;
+    }
+
+protected:
+    /**
+     * The next transaction in the queue
+     * This field is used for chaining transactions together into a queue.
+     *
+     * This field is not volatile because it is only accessed from within a
+     * critical section.
+     */
+    I2CTransaction * _next;
+    /// The target I2C address to communicate with
+    uint16_t _address;
+    /**
+     * The first I2CSegment in the transaction
+     *
+     * This field is not volatile because it is only accessed from within a
+     * critical section. This will still be valid if the critical section is
+     * replaced with an atomic operation.
+     */
+    detail::I2CSegment * _root;
+    /**
+     * The first I2CSegment in the transaction
+     *
+     * This is a helper field for building or processing I2C transactions.
+     * It allows the Transaction to easily locate the end of the queue when
+     * composing the transaction and, equally, the currently transferring segment
+     * while processing the transaction.
+     *
+     * This field is not volatile because it is only accessed from within a
+     * critical section.
+     */
+    detail::I2CSegment * _current;
+    /// The I2C frequency to use for the transaction
+    unsigned _hz;
+    /// Flag to indicate that the Transaction and its Segments were allocated with an irqsafe allocator
+    bool _irqsafe;
+    /// The I2C Object that launched this transaction
+    I2C * _issuer;
+    /// An array of I2C Event Handlers
+    detail::I2CEventHandler _handlers[I2C_TRANSACTION_NHANDLERS];
+};
+
+/** An I2C Master, used for communicating with I2C slave devices
+ *
+ * Example:
+ * @code
+ * // Read from I2C slave at address 0x62
+ *
+ * #include "mbed.h"
```

Comment:
mbed-drivers/mbed.h ? No namespace for I2C . The code seems to be incomplete, address is missing, tx_done is not valid function.